### PR TITLE
:bug: fix multiple tags +error logs not send + tags documentation

### PR DESCRIPTION
**Pull request by darkweaver87**
Errors without the `.Msg("")` were not correctly flushed.

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -73,12 +73,12 @@ A mutating webhook for Kubernetes, pointing the images to a new location.`,
 
 		imageSwapPolicy, err := types.ParseImageSwapPolicy(cfg.ImageSwapPolicy)
 		if err != nil {
-			log.Err(err)
+			log.Err(err).Send()
```
Comment (project0) [inline cmd/root.go:76]: I would prefer having log.Err(err).Msg("parse image swap policy failed") instead of a empty error without context. (same for below)

Comment (estahn) [reply cmd/root.go:76]: @darkweaver87 I agree with @project0 ... This would allow to avoid using `Send()`

see https://github.com/rs/zerolog/blob/164f7aa1a64acb4ec010b7c0d94f6066780a3a91/event.go#L111

File: cmd/root.go [hunk 2]
```diff
@@ -73,12 +73,12 @@ A mutating webhook for Kubernetes, pointing the images to a new location.`,
...
 		}
 
 		imageCopyPolicy, err := types.ParseImageCopyPolicy(cfg.ImageCopyPolicy)
 		if err != nil {
-			log.Err(err)
+			log.Err(err).Send()
```
Comment (estahn) [inline cmd/root.go:81]:
```suggestion
			log.Err(err).Str("policy", cfg.ImageCopyPolicy).Msg("parsing image copy policy failed")
```

File: pkg/registry/ecr.go [hunk 3]
```diff
@@ -107,6 +107,7 @@ func (e *ECRClient) buildEcrTags() []*ecr.Tag {
 	ecrTags := []*ecr.Tag{}
 
 	for _, t := range e.tags {
+		t := t
```
Comment (estahn) [inline pkg/registry/ecr.go:110]: Why is this required?

File: pkg/webhook/image_swapper.go [hunk 4]
```diff
@@ -215,26 +215,26 @@ func (p *ImageSwapper) Mutate(ctx context.Context, ar *kwhmodel.AdmissionReview,
 				createRepoName := reference.TrimNamed(srcRef.DockerReference()).String()
 				log.Ctx(lctx).Debug().Str("repository", createRepoName).Msg("create repository")
 				if err := p.registryClient.CreateRepository(createRepoName); err != nil {
-					log.Err(err)
+					log.Err(err).Send()
```
Comment (estahn) [inline pkg/webhook/image_swapper.go:218]:
```suggestion
					log.Err(err).Str("repository", createRepoName).Msg("failed to create repository")
```

File: pkg/webhook/image_swapper.go [hunk 5]
```diff
@@ -215,26 +215,26 @@ func (p *ImageSwapper) Mutate(ctx context.Context, ar *kwhmodel.AdmissionReview,
...
 				}
 
 				// Retrieve secrets and auth credentials
 				imagePullSecrets, err := p.imagePullSecretProvider.GetImagePullSecrets(pod)
 				if err != nil {
-					log.Err(err)
+					log.Err(err).Send()
```
Comment (estahn) [inline pkg/webhook/image_swapper.go:224]:
```suggestion
					log.Err(err).Msg("failed to retrieve image pull secrets from provider")
```

File: pkg/registry/ecr.go (hunk 3 above)
Comment (darkweaver87) [reply pkg/registry/ecr.go:110]: If we have multiple tags, then as it's a pointer we will add the same tag multiple times. I added tests for this. You can try without this line, you'll see it failing :-)

File: pkg/registry/ecr.go [hunk 6]
```diff
@@ -107,6 +107,7 @@ func (e *ECRClient) buildEcrTags() []*ecr.Tag {
...
 		tag := ecr.Tag{Key: &t.Key, Value: &t.Value}
```
Comment (project0) [inline pkg/registry/ecr.go:111]: @darkweaver87 rather do this:
```suggestion
		tag := ecr.Tag{Key: aws.String(t.Key), Value: aws.String(t.Value)}
```

File: .k8s-image-swapper.yml [hunk 7]
```diff
@@ -46,6 +46,8 @@ target:
       tags:
         - key: CreatedBy
           value: k8s-image-swapper
+        - key: AnotherTag
+          value: k8s-image-swapper
```
Comment (project0) [inline .k8s-image-swapper.yml:50]: maybe we should set a different value to catch race condition of key is different but the value is not correctly set?

File: pkg/registry/ecr.go (hunk 3 above)
Comment (project0) [reply pkg/registry/ecr.go:110]: this should not be required anymore when `aws.String()` is used.

**Review by estahn — APPROVED** [review]

**Review by project0 — APPROVED** [review]

File: pkg/webhook/image_swapper.go [hunk 8]
```diff
@@ -215,28 +215,26 @@ func (p *ImageSwapper) Mutate(ctx context.Context, ar *kwhmodel.AdmissionReview,
...
-					defer func() {
-						if err := os.RemoveAll(authFile.Name()); err != nil {
-							log.Err(err)
-						}
-					}()
-				}
-
 				if err != nil {
-					log.Err(err)
+					log.Err(err).Msg("generating authFile")
```
Comment (estahn) [inline pkg/webhook/image_swapper.go:229]:
```suggestion
					log.Err(err).Msg("failed generating authFile")
```

**Comment by estahn** [comment]
@darkweaver87 2 changes to the log messages, just to keep it consistent. Otherwise it looks good.

**Comment by darkweaver87** [comment]
OK, thanks :-)
